server/services/referral.service.js

```diff
@@ -355,44 +363,32 @@ export const updateReferralRewardStatus = async ({ referralId, offerId }) => {
     rewardStatus = REWARD_STATUS.PAYMENT_COMPLETED;
   }
 
-  await Referral.findByIdAndUpdate(
-    referralId,
-    { $set: { offerId, 'reward.status': rewardStatus } },
-    { new: true },
-  );
-};
-
-export const getReferralByOfferId = async (offerId) =>
-  Referral.findOne({ offerId: ObjectId(offerId) });
-
-export const updateAssignedReferral = async ({ referralId, transactionId, amountPaid }) => {
   const referral = await getReferralBasicInfoById(referralId).catch((error) => {
     throw new ErrorHandler(httpStatus.INTERNAL_SERVER_ERROR, 'Internal Server Error', error);
   });
 
   const user = await getUserById(referral.referrerId);
 
   const percentage = user.additionalInfo.referralPercentage;
-  const amountPayable = (percentage / 100) * amountPaid;
-  const total =
-    amountPayable +
-    referral.assignedReferral.transactions.reduce(
-      (accum, transaction) => accum + transaction.amountPayable,
-      0,
-    );
+  const amount = (percentage / 100) * amountPaid;
+  const previousTotal = referral.accumulatedReward.transactions.reduce(
+    (acc, transaction) => acc + transaction.amount,
+    0,
+  );
+  const total = amount + previousTotal;
 
-  const newAssignedReferral = {
+  const newAccumulatedReward = {
```

Comment:
I think we can leave this as `accumulatedReward`.

```suggestion
  const accumulatedReward = {
```